src/Microsoft.AspNetCore.Server.IIS/HttpProtocol.cs

```diff
@@ -313,13 +343,15 @@ public unsafe void CreateResponseHeader(bool appCompleted)
 
             HttpApiTypes.HTTP_RESPONSE_V2* pHttpResponse = NativeMethods.http_get_raw_response(_pHttpContext);
 
-            _pinnedHeaders = SetHttpResponseHeaders(pHttpResponse);
+            // We should be sending the headers here as there are many responses that don't have status codes.
+            var pinnedHeaders = SerializeHeaders(pHttpResponse);
```

Comment:
Uh what is this?

Reply:
Sorry, was working on another feature and decided to wait for another PR.